db/compacted_db_impl.h

```diff
@@ -25,11 +25,11 @@ class CompactedDBImpl : public DBImpl {
                      ColumnFamilyHandle* column_family, const Slice& key,
                      PinnableSlice* value) override;
   using DB::MultiGet;
-  virtual std::vector<Status> MultiGet(
-      const ReadOptions& options,
-      const std::vector<ColumnFamilyHandle*>&,
-      const std::vector<Slice>& keys, std::vector<std::string>* values)
-    override;
+  void MultiGet(const ReadOptions& options,
+                const std::vector<ColumnFamilyHandle*>&,
+                const std::vector<Slice>& keys,
+                std::vector<PinnableSlice>* values,
+                std::vector<Status>& statuses) override;
```

Comment:
non-const reference in function argument is not allowed in Google C++ Style.
Also, if we get a new MultiGet() API, we should allow ourselves to remove all the allocation inside. Now we are still allocating `values`.

CC @yoshinorim for this new MultiGet() API. Is it clear enough and will it allow MyRocks to use this function efficiently?